irohad/main/application.cpp

```diff
@@ -161,7 +163,14 @@ void Irohad::initStorage() {
       std::make_shared<shared_model::proto::ProtoPermissionToString>();
   auto block_converter =
       std::make_shared<shared_model::proto::ProtoBlockJsonConverter>();
-  auto block_storage_factory = std::make_unique<InMemoryBlockStorageFactory>();
+  auto block_storage_factory = std::make_unique<FlatFileBlockStorageFactory>(
+      []() {
+        return (boost::filesystem::temp_directory_path()
+                / boost::filesystem::unique_path())
+            .string();
+      },
+      block_converter,
+      log_manager_->getChild("FlatFileBlockFactory"));
```

Comment:
The `FlatFileBlockFactory` implementation itself does not use logging at all, so it does not need a logger. But the storages it creates do need one. So we should provide the factory with a logger which it will give the storage instances it creates.
```suggestion
      log_manager_->getChild("FlatFileBlockStorage")->getLogger());
```

Reply:
`FlatFileBlockStorageFactory` actually needs 2 loggers: one for `FlatFileBlockStorage` and another one for `FlatFile`. I don't like the idea to pass 2 loggers into it so a manager instance appeared.

Reply:
ok, that seems reasonable. than I would omit the `FlatFileBlockStorage` node of the hierarchy and pass `log_manager_` to the factory with the condition that it will not use it for its own logging, but only for creating loggers for `FlatFileBlockStorage` and `FlatFile`.